src/webui/webapplication.cpp

```diff
@@ -838,153 +855,598 @@ void WebApplication::action_command_setLocation()
     }
 }
 
-void WebApplication::action_command_rename()
+void WebApplication::torrentsRenameAction()
 {
-    CHECK_URI(0);
-    CHECK_PARAMETERS("hash" << "name");
+    checkParams({"hash", "name"});
 
-    QString hash = request().posts["hash"];
-    QString name = request().posts["name"].trimmed();
+    const QString hash = params()["hash"];
+    QString name = params()["name"].trimmed();
 
-    BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
-    if (torrent && !name.isEmpty()) {
-        name.replace(QRegularExpression("\r?\n|\r"), " ");
-        qDebug() << "Renaming" << torrent->name() << "to" << name;
-        torrent->setName(name);
+    if (name.isEmpty()) {
+        status(409, "Conflict");
+        print(tr("Incorrect torrent name"), Http::CONTENT_TYPE_TXT);
+        return;
     }
-    else {
-        status(400, "Incorrect torrent hash or name");
+
+    BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+    if (!torrent) {
+        status(404, "Not Found");
+        return;
     }
+
+    name.replace(QRegularExpression("\r?\n|\r"), " ");
+    qDebug() << "Renaming" << torrent->name() << "to" << name;
+    torrent->setName(name);
 }
 
-void WebApplication::action_command_setAutoTMM()
+void WebApplication::torrentsSetAutoManagementAction()
 {
-    CHECK_URI(0);
-    CHECK_PARAMETERS("hashes" << "enable");
+    checkParams({"enable"}, {"hashes"});
 
-    const QStringList hashes = request().posts["hashes"].split('|');
-    const bool isEnabled = parseBool(request().posts["enable"], false);
-    foreach (const QString &hash, hashes) {
-        BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
-        if (torrent)
+    const QStringList hashes = params()["hashes"].split('|');
+    const bool isEnabled = parseBool(params()["enable"], false);
+
+    if (!hashes.isEmpty()) {
+        for (const QString &hash : hashes) {
+            BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+            if (torrent)
+                torrent->setAutoTMMEnabled(isEnabled);
+        }
+    }
+    else {
```

Comment:
I see you made `hashes` optional here and I'm not sure this change is necessary... do you have specific use case?
I've no reason to object, just asking.

Reply:
Some methods can apply its action to one or more torrents. It's too expensive to send all hashes if you want to apply it to all torrents. Some of this methods have optimized alternatives earlier (e.g. pause/pauseAll). I start to transform it to uniform methods with optional "hashes" param.

Reply:
>It's too expensive to send all hashes if you want to apply it to all torrents. 

I agree.
Or how about use a special string for *all* operation? e.g. `hashes="all"` will change all torrent settings instead of omitting it? and also `hash` will become a required param.
Just an idea.

Reply:
Ok.
Please consider my additional commit. I've done it there (and some other improvements).

Reply:
>Please consider my additional commit. I've done it there (and some other improvements).

The result is pretty neat.
I can approve the current form after commit squash.